google_api_auth/src/yup_oauth2.rs

```diff
@@ -12,8 +14,8 @@ where
     }
 }
 
-struct YupAuthenticator<T> {
-    auth: Mutex<T>,
+struct YupAuthenticator<C> {
+    auth: Mutex<Authenticator<C>>,
```

Comment:
Authenticator is now Send+Sync I believe. Would you mind removing the now unnecessary Mutex?